pyproject.toml

```diff
@@ -53,6 +53,62 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
+alignment = [
+    "dtw-python",
+    "numba",
+]
+annotation = [
+    "attrs",
+    "hmmlearn",
+    "numba",
+    "pyod",
+]
+classification = [
+    "esig",
+    "mrsqm",
+    "numba",
+    "tensorflow",
+    "tsfresh",
+]
+clustering = [
+    "numba",
+    "tslearn",
+]
+forecasting = [
+    "hcrystalball",
+    "pmdarima",
+    "prophet",
+    "scikit-optimize",
+    "statsforecast",
+    "statsmodels",
+    "tbats",
+]
+networks = [
+    "tensorflow",
```

Comment:
thinking ahead, do we want to cater for multiple backends? There's `pytorch` and `tensorflow`, a given user is unlikely to want to install them both.

Reply:
Agreed, but will wait for comments from developers working on these.

What about `network-tensorflow` and `network-pytorch`?

Reply:
hm, makes sense! Although do we ever expect anything more than a single package in these?